client/lib/domains/is-dot-gay-notice-required.js

```diff
@@ -0,0 +1,7 @@
+import { find, get } from 'lodash';
+
+export function isDotGayNoticeRequired( productSlug, productsList ) {
+	const product = find( productsList, [ 'product_slug', productSlug ] ) || {};
+
+	return get( product, 'is_dot_gay_notice_required', false );
```

Comment:
Minor, but the calypso team has been working to remove lodash from everywhere, and in this case I think we can avoid it using plain JS.

```suggestion
	return product.?is_dot_gay_notice_required ?? false;
```